pkg/runner/flags.go

```diff
@@ -143,3 +158,40 @@ func ParseFlags(cfg *Config, args []string) (*Config, error) {
 	}
 	return cfg, nil
 }
+
+func validate(cfg *Config) error {
+	if err := api.Validate(&cfg.APICfg); err != nil {
+		return fmt.Errorf("error validating API configuration: %w", err)
+	}
+	if err := validateMultiTenancyFlags(cfg); err != nil {
+		return fmt.Errorf("error validating mlti-tenancy configuration: %w", err)
+	}
+	return nil
+}
+
+func validateMultiTenancyFlags(cfg *Config) error {
+	if !cfg.EnableMultiTenancy {
+		return nil
+	}
+	if cfg.validTenantsStr == allowAllTenants {
+		cfg.SkipTenantValidation = true
+		return nil
+	} else if cfg.validTenantsStr == "" {
+		return fmt.Errorf("'multi-tenancy-valid-tenants' cannot be empty")
+	}
+	cfg.ValidTenantsList = strings.Split(cfg.validTenantsStr, ",")
+	cfg.ValidTenantsList = removeEmptyTenants(cfg.ValidTenantsList)
+	return nil
+}
+
+// removeEmptyTenants protects against corner cases, when the user enters comma separated tenants
+// such that there is a trailing comma towards the end.
+func removeEmptyTenants(t []string) (tenants []string) {
```

Comment:
we should also make sure no tenant is equal to allowAllTenants

Reply:
If allowAllTenants is the option, then this function will not be called.

This takes care of it (first case inside `validateMultiTenancyFlags()`)

```golang
	if cfg.validTenantsStr == allowAllTenants {
		cfg.SkipTenantValidation = true
		return nil
	} else if cfg.validTenantsStr == "" {
		return fmt.Errorf("'multi-tenancy-valid-tenants' cannot be empty")
	}
```

Reply:
Ah, that is an interesting case. I definitely missed that!